lib/avro-patches/schema_compatibility/schema_compatibility.rb

```diff
@@ -117,17 +120,25 @@ def match_union_schemas(writers_schema, readers_schema)
         end
       end
 
+      # reader is a record
       def match_record_schemas(writers_schema, readers_schema)
-        writer_fields_hash = writers_schema.fields_hash
-        readers_schema.fields.each do |field|
-          if writer_fields_hash.key?(field.name)
-            return false unless full_match_schemas(writer_fields_hash[field.name].type, field.type)
-          else
-            return false unless field.default?
+        case writers_schema.type_sym
+        when :union
+          # find the first schema in the writer's union that matches the reader record's name
+          matching_union_member = writers_schema.schemas.select { |schema| schema.type_sym == :record && schema.name == readers_schema.name }.first
```

Comment:
Could this use `find` instead of `select {}.first`?

Reply:
changed